addons/agent_mirrorpeer_controller.go

```diff
@@ -57,37 +58,36 @@ type MirrorPeerReconciler struct {
 // For more details, check Reconcile and its Result here:
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
 func (r *MirrorPeerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	klog.Infof("Running MirrorPeer reconciler on spoke cluster")
-	// Fetch MirrorPeer for given Request
+	r.Logger.Info("Running MirrorPeer reconciler on spoke cluster")
+
 	var mirrorPeer multiclusterv1alpha1.MirrorPeer
 	err := r.HubClient.Get(ctx, req.NamespacedName, &mirrorPeer)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			klog.Info("Could not find MirrorPeer. Ignoring since object must have been deleted")
+			r.Logger.Info("MirrorPeer not found, ignoring since object must have been deleted", "MirrorPeer", req.NamespacedName.String())
 			return ctrl.Result{}, nil
 		}
-		klog.Error(err, "Failed to get MirrorPeer")
+		r.Logger.Error("Failed to retrieve MirrorPeer", "error", err)
 		return ctrl.Result{}, err
 	}
 
 	scr, err := utils.GetCurrentStorageClusterRef(&mirrorPeer, r.SpokeClusterName)
 	if err != nil {
-		klog.Error(err, "Failed to get current storage cluster ref")
+		r.Logger.Error("Failed to get current storage cluster ref", "error", err)
 		return ctrl.Result{}, err
 	}
 
 	agentFinalizer := r.SpokeClusterName + "." + SpokeMirrorPeerFinalizer
-
 	if len(agentFinalizer) > 63 {
 		agentFinalizer = fmt.Sprintf("%s.%s", r.SpokeClusterName[0:10], SpokeMirrorPeerFinalizer)
 	}
 
 	if mirrorPeer.GetDeletionTimestamp().IsZero() {
 		if !utils.ContainsString(mirrorPeer.GetFinalizers(), agentFinalizer) {
-			klog.Infof("Finalizer not found on MirrorPeer. Adding Finalizer %q", agentFinalizer)
+			r.Logger.Info("Adding finalizer to MirrorPeer", "finalizer", agentFinalizer)
 			mirrorPeer.Finalizers = append(mirrorPeer.Finalizers, agentFinalizer)
 			if err := r.HubClient.Update(ctx, &mirrorPeer); err != nil {
-				klog.Errorf("Failed to add finalizer to MirrorPeer %q", mirrorPeer.Name)
+				r.Logger.Error("Failed to add finalizer to MirrorPeer", "MirrorPeer", mirrorPeer.Name, "error", err)
```

Comment:
If request name and namespace is already logged we don't need MirrorPeer name and namespace. They're the same for MirrorPeerReconciler.

```suggestion
				r.Logger.Error("Failed to add finalizer to MirrorPeer", "error", err)
```

Reply:
done